src/main/scala/org/broadinstitute/dsde/agora/server/business/AgoraBusiness.scala

```diff
@@ -27,6 +27,22 @@ class AgoraBusiness(permissionsDataSource: PermissionsDataSource)(implicit ec: E
     }
   }
 
+  private def checkInsertPermission[T](db: DataAccess, agoraEntity: AgoraEntity, username: String, snapshots: Seq[AgoraEntity])(op: => ReadWriteAction[T]): ReadWriteAction[T] = {
+    // if previous snapshots exist, check ownership on them; if not, check permission on the namespace
+    if (snapshots.isEmpty)
+      checkNamespacePermission(db, agoraEntity, username, AgoraPermissions(Create))(op)
+    else {
+      val ownerPerm = AgoraPermissions(Manage)
```

Comment:
I specifically chose to not use `All` here. In practice, the Agora `All` permission maps to the FireCloud OWNER permission: https://github.com/broadinstitute/firecloud-orchestration/blob/develop/src/main/scala/org/broadinstitute/dsde/firecloud/core/AgoraPermissionHandler.scala#L60. Similarly, only FireCloud OWNERs will have Agora `Manage`. But if we ever went to a more granular approach, it seems like checking `Manage` is the right thing to do.